bin/tx-prover/src/proxy/mod.rs

```diff
@@ -74,77 +74,65 @@ impl LoadBalancer {
         Err(error)
     }
 
-    /// Remove the request ID from the corresponding worker queue
-    pub async fn remove_request_from_queue(request_id: &str) {
-        let mut ctx_guard = QUEUES.write().await;
-
-        // Get the worker by checking each queue
-        let worker = ctx_guard
-            .iter()
-            .find(|(_, queue)| queue.contains(&request_id.to_string()))
-            .map(|(worker, _)| worker.clone())
-            .expect("Worker not found");
-
-        // Remove the request ID from the queue for the specific worker
-        if let Some(worker_queue) = ctx_guard.get_mut(&worker) {
-            if !worker_queue.is_empty() {
-                worker_queue.remove(0);
-            }
-        }
+    pub async fn get_available_worker(&self) -> Option<Backend> {
+        self.available_workers.write().await.pop()
     }
 
-    /// Get the request ID from the session headers
-    pub fn get_request_id(session: &Session) -> Result<&str> {
-        session
-            .get_header("X-Request-ID")
-            .ok_or(Error::new(ErrorType::InternalError))?
-            .to_str()
-            .map_err(|_| Error::new(ErrorType::InternalError))
+    pub async fn set_available_worker(&self, worker: Backend) {
+        self.available_workers.write().await.push(worker);
```

Comment:
Using `HashSet` doesn't sound like a much better option, since to pop a Worker I will need to create an iterator each time. I'm adding a `contains` call to check if already exists the worker in the list of available and panic if it is already there.